File: Node/TypeORM_ORM/src/bll/services/customerService.ts
import { Customers } from 'dal/entities/Customers.js';
import { CustomerRepository } from '../../dal/repositories/customerRepository';

export class CustomerService {
  private categoryRepository = new CustomerRepository();

  async getAll() {
    return await this.categoryRepository.getAll();
  }

  async getById(id: string) {
    return await this.categoryRepository.getById(id);
  }

  async create(data: Partial<Customers>) {
    return await this.categoryRepository.create(data);
  }

  async update(id: string, data: Partial<Customers>) {
    return await this.categoryRepository.update(id, data);
  }

  async delete(id: string) {
    return await this.categoryRepository.delete(id);
  }
}

File: Go/SQLBoiler_ORM/bll/services/customerService.go
package services

import (
	"SQLBOILER_ORM/dal/entities"
	"SQLBOILER_ORM/dal/repositories"
	"context"
)

type CustomerService struct {
	repo *repositories.CustomerRepository
}

func NewCustomerService(repo *repositories.CustomerRepository) *CustomerService {
	return &CustomerService{repo: repo}
}

func (s *CustomerService) GetByID(ctx context.Context, id string) (*entities.Customer, error) {
	customer, err := s.repo.GetByID(ctx, id)
	if err != nil {
		return nil, err
	}
	return customer, nil
}

func (s *CustomerService) GetAll(ctx context.Context) ([]*entities.Customer, error) {
	return s.repo.GetAll(ctx)
}

func (s *CustomerService) Create(ctx context.Context, customer *entities.Customer) (*entities.Customer, error) {
	return s.repo.Create(ctx, customer)
}

func (s *CustomerService) Update(ctx context.Context, customer *entities.Customer) error {
	return s.repo.Update(ctx, customer)
}

func (s *CustomerService) Delete(ctx context.Context, id string) error {
	return s.repo.Delete(ctx, id)
}
